packages/react-reconciler/src/ReactFiber.js

```diff
@@ -546,7 +548,15 @@ export function resetWorkInProgress(
     workInProgress.child = null;
     workInProgress.memoizedProps = null;
     workInProgress.memoizedState = null;
-    workInProgress.updateQueue = null;
+    // updateQueue must never be null on ClassComponent or HostRoot
```

Comment:
In line with https://github.com/facebook/react/blob/c5d2fc7127654e43de59fff865b74765a103c4a5/packages/react-reconciler/src/ReactUpdateQueue.js#L399

would it make more sense to move this logic into a `resetUpdateQueue` inside `ReactUpdatedQueue.js`?